node/runtime/src/lib.rs

```diff
@@ -346,6 +354,25 @@ impl Runtime {
     }
 }
 
+pub struct StateDbViewer {
```

Comment:
hm, what is this object suppose to mean?
why is State has blockchain in it?

Reply:
this is the minimal struct needed for responding to view calls. it needs a blockchain because it needs to know the latest root state.

Reply:
We should pass merkle root from whoever is responsible outside of the runtime. E.g. runtime should not depend on blockchain at all.

Reply:
ok, we can move this elsewhere, or rename it , but seemed like a reasonable place to keep it for now. this is basically just a way to get this functionality with the lightest weight wrapper around it. it may make a bit more sense with this change (https://github.com/nearprotocol/nearcore/pull/140/files#diff-5e5e1c3aec9ddfde0a9054d062ab3db9R350)